pulsar-broker/src/main/java/org/apache/pulsar/utils/auth/tokens/TokensCliUtils.java

```diff
@@ -0,0 +1,231 @@
+    public static class CommandCreateSecretKey {
+        @Parameter(names = { "-a",
+                "--signature-algorithm" }, description = "The signature algorithm for the new secret key.")
+        SignatureAlgorithm algorithm = SignatureAlgorithm.HS256;
+
+        @Parameter(names = { "-o",
+                "--output" }, description = "Write the secret key to a file instead of stdout")
+        String outputFile;
+
+        @Parameter(names = {
+                "-b", "--base-64" }, description = "Encode the key in base64")
+        boolean base64 = false;
+
+        public void run() throws IOException {
+            SecretKey secretKey = AuthTokenUtils.createSecretKey(algorithm);
+            byte[] encoded = secretKey.getEncoded();
+
+            if (base64) {
+                encoded = Encoders.BASE64.encode(encoded).getBytes();
+            }
+
+            if (outputFile != null) {
+                Files.write(Paths.get(outputFile), encoded);
+            } else {
+                System.out.write(encoded);
+            }
+        }
+    }
+
+    @Parameters(commandDescription = "Create a new or pair of keys public/private")
+    public static class CommandCreateKeyPair {
+        @Parameter(names = { "-a",
+                "--signature-algorithm" }, description = "The signature algorithm for the new key pair.")
+        SignatureAlgorithm algorithm = SignatureAlgorithm.RS256;
+
+        @Parameter(names = {
+                "--output-private-key" }, description = "File where to write the private key", required = true)
+        String privateKeyFile;
+        @Parameter(names = {
+                "--output-public-key" }, description = "File where to write the public key", required = true)
+        String publicKeyFile;
+
+        public void run() throws IOException {
+            KeyPair pair = Keys.keyPairFor(algorithm);
+
+            Files.write(Paths.get(publicKeyFile), pair.getPublic().getEncoded());
+            Files.write(Paths.get(privateKeyFile), pair.getPrivate().getEncoded());
+        }
+    }
+
+    @Parameters(commandDescription = "Create a new token")
+    public static class CommandCreateToken {
+
+        @Parameter(names = { "-s",
+                "--subject" }, description = "Specify the 'subject' or 'principal' associate with this token", required = true)
+        private String subject;
+
+        @Parameter(names = { "-e",
+                "--expiry-time" }, description = "Relative expiry time for the token (eg: 1h, 3d, 10y). (m=minutes) Default: no expiration")
+        private String expiryTime;
+
+        @Parameter(names = { "-pk",
+                "--is-private-key" }, description = "Indicate the signing key is a private key (rather than a symmetric secret key)")
+        private Boolean isPrivateKey = false;
+
+        @Parameter(names = { "-k",
+                "--signing-key" }, description = "Pass the signing key. This can either be: data:, file:, etc..", required = true)
```

Comment:
```--signing-key``` is a new concept when you compare it the "create" commands. it would be better to have
```--private-key``` and ```--secret-key```, require that only one of them is specified, and kill the ```--is-private-key``` flag.

Reply:
Especially since there's no concept of signing key in the broker conf.

Reply:
Good point, fixed